src/client.rs

```diff
@@ -138,11 +138,9 @@ fn validate_sequence_set(
 #[derive(Debug)]
 pub struct Session<T: Read + Write> {
     conn: Connection<T>,
-    pub(crate) unsolicited_responses_tx: mpsc::Sender<UnsolicitedResponse>,
-
     /// Server responses that are not related to the current command. See also the note on
     /// [unilateral server responses in RFC 3501](https://tools.ietf.org/html/rfc3501#section-7).
-    pub unsolicited_responses: mpsc::Receiver<UnsolicitedResponse>,
+    pub unsolicited_responses: VecDeque<UnsolicitedResponse>,
```

Comment:
In case we want to change this again in the future, I wonder if we should stop having this be a `pub` field, and instead have a `take_unsolicited()` method, or something like it? Maybe:
```rust
impl Session {
    pub fn next_unsolicited(&mut self) -> Option<UnsolicitedResponse>;
    pub fn all_unsolicited(&mut self) -> Vec<UnsolicitedResponse>;
}
```
That way the exact type we use here isn't exposed in the API, and thus can be changed. What do you think?

Reply:
Maybe even:
```rust
    pub fn all_unsolicited(&mut self) -> impl Iterator<Item = UnsolicitedResponse> {
        std::mem::take(&mut self.unsolicited_responses).into_iter()
    }
```
Moving out of the field safes us an additional allocation of a `Vec`, while still hiding the concrete queue-type from the public interface.

Reply:
That works too, though maybe `-> impl ExactSizeIterator` to be even nicer to the caller :)